src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -1106,12 +1106,13 @@ private bool HaveAllDependentServicesStopped(ICollection<ServiceController> depe
         /// <param name="services">A list of services.</param>
         internal void RemoveNotStoppedServices(List<ServiceController> services)
         {
-            foreach (ServiceController service in services)
+            for (int i = services.Count - 1; i >= 0; i--)
```

Comment:
Is there a reason you're enumerating from the end to the beginning here?

Also, I don't see a real need to use a for loop here. For loops are nice, but tend to create confusing and less readable code on the whole.

Perhaps a better solution would be to attempt to stop each service, then filter out the stopped services with one of the List filter methods.

Reply:
> Is there a reason you're enumerating from the end to the beginning here?

Enumerating from the end tends to move fewer items.

> Also, I don't see a real need to use a for loop here. For loops are nice, but tend to create confusing and less readable code on the whole.

Sounds good. Updated to use `RemoveAll` (and `TrueForAll` for another method BTW).

> Perhaps an alternative solution would be to attempt to stop each service

The services have been requested to stop before this filter method.